components/eam/src/physics/crm/cloud_diagnostics.F90

```diff
@@ -1,525 +1,374 @@
 
 module cloud_diagnostics
-
-!---------------------------------------------------------------------------------
-! Purpose:
-!
-! Put cloud physical specifications on the history tape
-!  Modified from code that computed cloud optics
-!
-! Author: Byron Boville  Sept 06, 2002
-!  Modified Oct 15, 2008
-!    
-!
-!---------------------------------------------------------------------------------
-
-   use shr_kind_mod,  only: r8=>shr_kind_r8
-   use ppgrid,        only: pcols, pver,pverp
-   use physconst,     only: gravit
-   use cam_history,   only: outfld
-   use cam_history,   only: addfld, horiz_only, add_default
-
-   implicit none
-   private
-   save
-
-   public :: cloud_diagnostics_init
-   public :: cloud_diagnostics_calc
-   public :: cloud_diagnostics_register
-
-! Local variables
-   integer :: dei_idx, mu_idx, lambda_idx, iciwp_idx, iclwp_idx, cld_idx  ! index into pbuf for cloud fields
-   integer :: ixcldice, ixcldliq, rei_idx, rel_idx
-
-   logical :: do_cld_diag, mg_clouds, rk_clouds, camrt_rad
-   integer :: conv_water_in_rad
+!-------------------------------------------------------------------------------
+! Purpose: Put cloud physical specifications on the history tape
+!-------------------------------------------------------------------------------
+  use shr_kind_mod,  only: r8=>shr_kind_r8
+  use ppgrid,        only: begchunk, endchunk, pcols, pver,pverp
+  use physconst,     only: gravit
+  use cam_history,   only: outfld
+  use cam_history,   only: addfld, horiz_only, add_default
+
+  implicit none
+  private
+  save
+
+  public :: cloud_diagnostics_register
+  public :: cloud_diagnostics_init
+  public :: cloud_diagnostics_calc
    
-   integer :: cicewp_idx = -1
-   integer :: cliqwp_idx = -1
-   integer :: cldemis_idx = -1
-   integer :: cldtau_idx = -1
-   integer :: nmxrgn_idx = -1
-   integer :: pmxrgn_idx = -1
+  ! Local variables
+  integer :: dei_idx, mu_idx, lambda_idx, iciwp_idx, iclwp_idx, cld_idx  ! index into pbuf for cloud fields
+  integer :: ixcldice, ixcldliq, rei_idx, rel_idx
 
-   ! Index fields for precipitation efficiency.
-   integer :: acpr_idx, acgcme_idx, acnum_idx
+  logical :: do_cld_diag
+  integer :: conv_water_in_rad
 
+  integer :: cicewp_idx = -1
+  integer :: cliqwp_idx = -1
 
-contains
+  ! Index fields for precipitation efficiency.
+  integer :: acpr_idx, acgcme_idx, acnum_idx
 
+contains
+!===============================================================================
 !===============================================================================
-  subroutine cloud_diagnostics_register
-
-    use phys_control,  only: phys_getopts
-    use physics_buffer,only: pbuf_add_field, dtype_r8, dtype_i4
 
-    character(len=16) :: rad_pkg, microp_pgk
+subroutine cloud_diagnostics_register
+  ! use phys_control,  only: phys_getopts
```

Comment:
Can we remove these commented out lines?

Reply:
yup